corehq/apps/enterprise/models.py

```diff
@@ -53,11 +64,29 @@ def get_domains(cls, source_domain):
             return []
         return list(set(config.domains) - {config.source_domain})
 
+    @classmethod
+    @quickcache(['source_domain'], timeout=7 * 24 * 60 * 60)
+    def is_source_domain(cls, source_domain):
+        """
+        Returns true if given domain is the source domain for an enabled configuration.
+        """
+        try:
+            cls.objects.get(is_enabled=True, source_domain=source_domain)
+        except cls.DoesNotExist:
+            return False
+        return True
+
     def save(self, *args, **kwargs):
+        self.is_source_domain.clear(self.__class__, self.source_domain)
+        for domain in self.account.get_domains():
+            self.get_domains.clear(self.__class__, domain)
+            self.get_source_domain.clear(self.__class__, domain)
+
         super().save(*args, **kwargs)
+        self.is_source_domain.clear(self.__class__, self.source_domain)
         for domain in self.account.get_domains():
             self.get_domains.clear(self.__class__, domain)
-            self.get_by_domain.clear(self.__class__, domain)
+            self.get_source_domain.clear(self.__class__, domain)
```

Comment:
Is this related to race conditions and worrying about the cache being populated with stale data? I'm struggling to remember why we would need to clear the cache beforehand, since until the object is saved successfully, it isn't necessary to clear the cache right?

Reply:
My thought was around clearing the caches for both the old domains and the new domains. It's true that clearing isn't necessary until the object is saved successfully, this was just more concise than storing the old values.